src/AssemblerHttpServer-Alignments.cpp

```diff
@@ -1182,6 +1183,255 @@ void Assembler::computeAllAlignments(
 }
 
 
+void Assembler::sampleReads(vector<OrientedReadId>& sample, uint64_t n){
+    sample.clear();
+
+    while (sample.size() < n) {
+        // Randomly select a read in the read set
+        uint16_t readId = uint16_t(rand() % reads.size());
+
+        // Randomly select an orientation
+        uint16_t strand = uint16_t(rand() % 2);
+
+        const OrientedReadId r(readId, strand);
+
+        sample.push_back(r);
```

Comment:
[Nit-pick] 
```
const OrientedReadId r(readId, strand);
sample.push_back(r);
```
creates an object and copies it into the vector. I would assume that the compiler is smart enough to optimize this away. 

You could also do the following to use C++ move semantics explicitly (which will move the object into the vector instead of copying it).

```
sample.push_back(OrientedReadId(readId, strand));
```

Feel free to ignore this comment if you like your style. I am 90% certain the compiler will do the right thing here.

Reply:
For this one I am aware that it would copy without optimization, but as you said there is no point in using `emplace_back`, `mov`, or a one-liner to try to manually optimize it because the compiler will do it.

Reply:
The compiler will do the right optimization here, and both ways are clear and readable, although I personally prefer @bagashe's way  because it does not introduce an extra named variable. In any event most of the cost by far will go in computing the alignments, so the performance of these preparation steps does not really matter.